src/internal/connector/discovery/discovery.go

```diff
@@ -69,6 +69,22 @@ func Users(
 	return users, nil
 }
 
+// UserDetails fetches detailed info like - userPurpose for all users in the tenant.
+func UserDetails(
```

Comment:
Rather than splitting it out, packing this info UserInfo would make for a better consolidated owner.

Reply:
Done that. Just a thought- 
In cases where only HasMailBox info was required, we make two more calls. Which might be a overhead.

Reply:
Good point. How about exposing a func that can make that call in isolation?